emalloc.c

```diff
@@ -257,7 +257,7 @@ static block_t* neighbor_right(block_t* me)
 static block_t* possibly_merge(block_t* b)
 {
     block_t* nr = neighbor_right(b);
-    while (nr && is_alloced(nr))
+    if (nr && !is_alloced(nr))
```

Comment:
Thanks for spotting this.
Condition is correct but 'if' should still be 'while' so it keeps merging free blocks from the right neighbor.

Reply:
also need update nr :-)

Reply:
I believe that the `while` statement is unnecessary and the `if` statement is more suitable. Since the process of merging operations is performed each time, it is impossible that a specified block would have two consecutive freed blocks.

Reply:
@ClawSeven I think it makes sense to me.

I found another potential issue: https://github.com/intel/sgx-emm/blob/eb4d16c459a27b1aa4e77fa4b2104dc537e2d001/emalloc.c#L244
Seems it should be `mm_reserve_t* r1 = find_used_in_reserve((size_t)me, block_size(me));`

@haitaohuang may need your confirmation.

Reply:
@xxu36 yes. Good catch.
But I still think while loop is needed because we never merge to the left. So there could be scenarios where UUU ->UFU->UFF->FFF where U means used block, F means freed.